assets/src/domain/shared/services/predicates/listPredicates.ts

```diff
@@ -0,0 +1,7 @@
+import { pluck } from 'ramda';
+
+import { Cacheable, Entity } from '@appServices/apollo/types';
+
+export const getGuids = pluck<keyof Pick<Entity, 'id'>>('id');
+
+export const getCacheIds = pluck<keyof Pick<Cacheable, 'cacheId'>>('cacheId');
```

Comment:
this is amazing, so much boilerplate code has been avoided here :taco:

Reply:
ok, not that much: `entity.map({id} => id)`
but anyway `pluck` is more cleaner and it has some interesting use cases we might use in the future:
```
R.pluck(0, [[1, 2], [3, 4]]);               //=> [1, 3]
R.pluck('val', {a: {val: 3}, b: {val: 5}}); //=> {a: 3, b: 5}
```

Reply:
I'm guessing that the way Manzoor has done this preserves the Types better than simply mapping a prop over to the returned array, so consuming code will know that it is specifically receiving an array of GUIDs or an array of CacheIDs as opposed to an array with some entity prop in it.